Source/CarthageKit/Xcode.swift

```diff
@@ -818,15 +825,24 @@ public func buildInDirectory( // swiftlint:disable:this function_body_length
 			}
 			.collectTaskEvents()
 			.flatMapTaskEvents(.concat) { (urls: [URL]) -> SignalProducer<(), CarthageError> in
-				guard let dependency = dependency else {
-					return .empty
-				}
+				let isCurrentProject = dependency == nil
 
-				return createVersionFile(
-					for: dependency.dependency, version: dependency.version,
-					platforms: options.platforms, buildProducts: urls, rootDirectoryURL: rootDirectoryURL
-				)
-				.flatMapError { _ in .empty }
+				if isCurrentProject {
+					return createVersionFileForCurrentProject(platforms: options.platforms,
+							buildProducts: urls,
+							rootDirectoryURL: rootDirectoryURL
+						)
+						.flatMapError { _ in .empty }
+				} else {
+					return createVersionFile(
+						for: dependency!.dependency,
```

Comment:
Instead of checking `== nil` above and then using `!` here, I'd prefer to use `if let` or `guard let`.

Reply:
Yes, sorry. I missed it.